### PR TITLE
Add to Wishlist

### PR DESCRIPTION
**Pull request by SaurabhChawda**
Added a wishlist functionality in Trendy-t E-Commerce. Whenever the user clicks on the Wishlist button, It will add to the wishlist page. Users can remove the Cart from the wishlist page.
Screenshot (288)

### CODE REVIEW

File: src/Contexts/UserContext.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext, useReducer } from "react";
+import { Reducer } from "../Reducer/userReducer";
+const UserContext = createContext(null);
+
+const UserProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(Reducer, { wishlist: [], cart: [] });
+  // console.log(`WishList:- ${state.wishlist}`);
```
Comment (shaksham08) [inline src/Contexts/UserContext.js:7]: Remove comments if not needed

**Review by shaksham08 — APPROVED** [review]
Code looks clean , just added one comment

File: src/Reducer/userReducer.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          wishlist: state.wishlist
+            .filter((item) => item.id !== action.id)
+            .concat(action.payload),
+        };
+      }
+    case "Remove_From_Wishlist":
+      return {
+        ...state,
+        wishlist: state.wishlist.filter((item) =>
+          item.id === action.payload ? false : true
```
Comment (JantuDeb) [inline src/Reducer/userReducer.js]: `filter(item=>item.id===action.payload)` will work fine,  no need to return `true` or `false` explicitly

File: src/Reducer/userReducer.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            .filter((item) => item.id !== action.id)
+            .concat({ ...action.payload, qty: 1 }),
+        };
+      }
+    case "Remove_From_Cart":
+      return {
+        ...state,
+        cart: state.cart.filter((item) =>
+          item.id === action.payload ? false : true
+        ),
```
Comment (JantuDeb) [inline src/Reducer/userReducer.js]: same here

**Review by JantuDeb — APPROVED** [review]
left few comments

File: src/Contexts/UserContext.js (hunk 1 above)
Comment (SaurabhChawda) [reply src/Contexts/UserContext.js:7]: I will change, Thankyou

File: src/Reducer/userReducer.js (hunk 3 above)
Comment (SaurabhChawda) [reply src/Reducer/userReducer.js]: updated, Thankyou

**Pull request merged**